aiida/backends/djsite/globalsettings.py

```diff
@@ -11,22 +11,24 @@
 Functions to manage the global settings stored in the DB (in the DbSettings
 table.
 """
-
+from django.db import IntegrityError
+from aiida.common.exceptions import UniquenessError
 
 
 def set_global_setting(key, value, description=None):
     """
-    Set a global setting in the DbSetting table (therefore, stored at the DB
-    level).
+    Set a global setting in the DbSetting table (therefore, stored at the DB level).
     """
     from aiida.backends.djsite.db.models import DbSetting
 
     # Before storing, validate the key
     DbSetting.validate_key(key)
 
-    # This also saves in the DB    
-    DbSetting.set_value(key, value,
-                        other_attribs={"description": description})
+    # This also saves in the DB
+    try:
+        DbSetting.set_value(key, value, other_attribs={"description": description})
+    except IntegrityError as exception:
+        raise UniquenessError(exception)
```

Comment:
Did you try to generate this case? Normally if there is an integrity error and you don't do a rollback the database is not usable and the next SQL command will crash I believe.
Typically you create a checkpoint before and, in the exception, you rollback (or you commit if all went well), it's a pattern done also in other parts of the code and IIRC it is needed for Django+PSQL.

Reply:
I modified your test scripts to call the new `set_process_state_change_timestamp` for a dummy `Process` and without this guard, as expected I got the `IntegrityError`. Putting in the guard fixed that problem and I didn't observe any problems with database state. Notably, I did not have to change the global setter for SqlAlchemy. Did you expect to behave differently for SqlA?

Reply:
You are right, I tried to perform some queries and commits in Django and after the exception all was still working fine. I think it would be worth trying also in SQLA though (i.e. in after catching the `except UniquenessError`, trying to do a query for instance).
Otherwise, only my other comment remains.

Reply:
I performed the exact same experiment in Sqla (without the guard at first) and the exception was there never even raised. To be honest I did not investigate in depth why not, maybe it is dealt with deeper down to prevent this from happening. So that is why I decided to not change the implementation of `set_global_setting` in `globalsettings.py` for SqlAlchemy.